src/assisted_installer_controller/assisted_installer_controller_test.go

```diff
@@ -777,6 +711,89 @@ var _ = Describe("installer HostRoleMaster role", func() {
 			callUploadLogs(250 * time.Millisecond)
 		})
 	})
+
+	Context("waitingForClusterVersion", func() {
+		tests := []struct {
+			name             string
+			currentCVOStatus OperatorStatus
+			newCVOStatus     OperatorStatus
+			shouldSendUpdate bool
+		}{
+			{
+				name:             "(false, no message) -> (false, no message)",
+				currentCVOStatus: OperatorStatus{isAvailable: false, message: ""},
+				newCVOStatus:     OperatorStatus{isAvailable: false, message: ""},
+				shouldSendUpdate: false,
+			},
+			{
+				name:             "(false, no message) -> (false, with message)",
+				currentCVOStatus: OperatorStatus{isAvailable: false, message: ""},
+				newCVOStatus:     OperatorStatus{isAvailable: false, message: "message"},
+				shouldSendUpdate: true,
+			},
+			{
+				name:             "(false, with message) -> (false, same message)",
+				currentCVOStatus: OperatorStatus{isAvailable: false, message: "message"},
+				newCVOStatus:     OperatorStatus{isAvailable: false, message: "message"},
+				shouldSendUpdate: false,
+			},
+			{
+				name:             "(false, with message) -> (false, new message)",
+				currentCVOStatus: OperatorStatus{isAvailable: false, message: "message"},
+				newCVOStatus:     OperatorStatus{isAvailable: false, message: "new"},
+				shouldSendUpdate: true,
+			},
+			{
+				name:             "(false, with message) -> (false, no message)",
+				currentCVOStatus: OperatorStatus{isAvailable: false, message: "message"},
+				newCVOStatus:     OperatorStatus{isAvailable: false, message: ""},
+				shouldSendUpdate: false,
+			},
+			{
+				name:             "(false, no message) -> (true, no message)",
+				currentCVOStatus: OperatorStatus{isAvailable: false, message: ""},
+				newCVOStatus:     OperatorStatus{isAvailable: true, message: ""},
+				shouldSendUpdate: true,
+			},
```

Comment:
tests look great.
I know `isAvailable` take precedence [here](https://github.com/openshift/assisted-installer/pull/208/files#diff-1da579fda3bc689940661d48dd744a46f23824270427207cfae4f8c7448eddd2R561), which is good, but we should also check for: `"(false, no message) -> (true, with message)"`